src/main/java/com/blamejared/crafttweaker/impl/recipes/handlers/vanilla/StoneCutterRecipeHandler.java

```diff
@@ -0,0 +1,45 @@
+package com.blamejared.crafttweaker.impl.recipes.handlers.vanilla;
+
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.api.recipes.IRecipeHandler;
+import com.blamejared.crafttweaker.api.recipes.ReplacementRule;
+import com.blamejared.crafttweaker.api.util.StringUtils;
+import com.blamejared.crafttweaker.impl.item.MCItemStackMutable;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.StonecuttingRecipe;
+
+import java.util.List;
+import java.util.Optional;
+
+@IRecipeHandler.For(StonecuttingRecipe.class)
+public final class StoneCutterRecipeHandler implements IRecipeHandler<StonecuttingRecipe> {
+    
+    @Override
+    public String dumpToCommandString(final IRecipeManager manager, final StonecuttingRecipe recipe) {
+        
+        return String.format(
+                "stoneCutter.addRecipe(%s, %s, %s);",
+                StringUtils.quoteAndEscape(recipe.getId()),
+                new MCItemStackMutable(recipe.getRecipeOutput()).getCommandString(),
+                IIngredient.fromIngredient(recipe.getIngredients().get(0)).getCommandString()
+        );
+    }
+    
+    @Override
+    public Optional<StonecuttingRecipe> replaceIngredients(final IRecipeManager manager, final StonecuttingRecipe recipe, final List<ReplacementRule> rules) {
+        
+        final Ingredient originalInput = recipe.getIngredients().get(0);
+        /*mutable*/ Ingredient input = originalInput;
+        for (final ReplacementRule rule : rules) {
+            if (rule.shouldReplace(input)) {
+                input = rule.getVanillaTo();
+            }
+        }
+    
+        if (originalInput == input) return Optional.empty();
```

Comment:
It seems like this logic is used in quite a few of these handlers, maybe it makes sense to move it into an own method?

Maybe something like
```java
protected function replaceSingleIngredient(Ingredient originalInput, List<ReplacementRule> rules) {
    /*mutable*/ Ingredient input = originalInput;
    for (final ReplacementRule rule : rules) {
        if(rule.shouldReplace(input) {
            input = rule.getVanillaTo();
        }
    }
    return input;
}
```